libunwind/src/DwarfInstructions.hpp

```diff
@@ -381,24 +381,22 @@ typename A::pint_t
 DwarfInstructions<A, R>::evaluateExpression(pint_t expression, A &addressSpace,
                                             const R &registers,
                                             pint_t initialStackValue) {
-  const bool log = false;
   pint_t p = expression;
   pint_t expressionEnd = expression + 20; // temp, until len read
   pint_t length = (pint_t)addressSpace.getULEB128(p, expressionEnd);
   expressionEnd = p + length;
-  if (log)
-    fprintf(stderr, "evaluateExpression(): length=%" PRIu64 "\n",
-            (uint64_t)length);
+  _LIBUNWIND_TRACE_DWARF_EVAL("evaluateExpression(): length=%" PRIu64 "\n",
+                              (uint64_t)length);
   pint_t stack[100];
   pint_t *sp = stack;
   *(++sp) = initialStackValue;
 
   while (p < expressionEnd) {
-    if (log) {
-      for (pint_t *t = sp; t > stack; --t) {
-        fprintf(stderr, "sp[] = 0x%" PRIx64 "\n", (uint64_t)(*t));
-      }
+#if _LIBUNWIND_TRACING_DWARF_EVAL
```

Comment:
Not attached to this line: do we want to remove the `<stdio.h>` include in this file then? But we'd need to include it from `config.h`?

Reply:
good question. I think it might be a good idea to completely remove `#include <stdio.h>` everywhere unless some logging is active. I can try to add that. Though maybe this should be a separate PR since I think if we do this, we should do this in the library as a whole? It seems to be included in a majority of the files. Though all those includes are kinda redundant since they all include `"config.h"` anyways, which already includes `<stdio.h>`. I guess we could just conditionally include `<stdio.h>` in `"config.h"` only when logging is active and remove the include everywhere else?

Reply:
Yeah, I think that sounds like the best approach to me (i.e. removing `<stdio.h>` everywhere except in `"config.h"`). You can do it in this patch or in a follow-up, I don't mind.

Reply:
Actually, there seems to be more of the same: https://github.com/llvm/llvm-project/blob/d97981c98a70ebeaa8901f34921b0a69a068ff5d/libunwind/src/UnwindCursor.hpp#L1714

I just overlooked that since that path wasn't being built in my setup. I'll update this patch to do the same thing for this code too and remove the `<stdio.h>` includes.

Reply:
We normally don't output like that from the tests, but I am not aware of any mechanism available from libunwind to conditionally output information. FWIW I think it would be reasonable for your patch not to change that test (even though it would be a barrier for porting the tests to bare metal).

Reply:
makes sense, I'll just leave the test as-is.